### PR TITLE
add request timeout warning for oidc flow

### PR DESCRIPTION
**Pull request by lgadban**
# Description

Update OIDC docs with warnings for request timeout

# Context

fixes https://github.com/solo-io/gloo/issues/2960

# Checklist:

N/A

### CODE REVIEW

File: docs/content/guides/security/auth/extauth/oauth/_index.md [hunk 1]
```diff
@@ -10,10 +10,14 @@ description: External Auth with Oauth
...
+The goal of OIDC is to address this ambiguity by additionally requiring Identity Providers to return a well-defined **ID Token**. OIDC ID tokens follow the [JSON Web Token]({{% versioned_link_path fromRoot="/guides/security/auth/jwt" %}}) standard and contain specific fields that your applications can expect and handle. This standardization allows you to switch between Identity Providers - or support multiple ones at the same time - with minimal, if any, changes to your downstream services; it also allows you to consistently apply additional security measures like _Role-based Access Control (RBAC)_ based on the identity of your users, i.e. the contents of their ID token (check out [this guide]({{% versioned_link_path fromRoot="/guides/security/auth/jwt/access_control" %}}) for an example of how to use Gloo to apply RBAC policies to JWTs).
```
Comment (kdorosh) [inline docs/content/guides/security/auth/extauth/oauth/_index.md:13]: we have the `protobuf` directive for exactly this use case. using it, if we ever move the reference location we wouldn't need to update this link.

sample usage: docs/content/guides/security/rate_limiting/simple/_index.md (see `{{% protobuf name="ratelimit.options.gloo.solo.io.IngressRateLimit" display="format"%}}`)

File: docs/content/guides/security/auth/extauth/oauth/_index.md [hunk 2]
```diff
@@ -10,10 +10,14 @@ description: External Auth with Oauth
...
 
 In this guide, we will focus on the format of the Gloo API for OIDC authentication.
 
+{{% notice warning %}}
+This feature requires Gloo's external auth server to communicate with an external OIDC provider/authorization server. Because of this interaction, the OIDC flow may take longer than the default timeout of 200ms. You can increase this timeout by setting the [`requestTimeout` value on external auth settings]({{% versioned_link_path fromRoot="/reference/api/github.com/solo-io/gloo/projects/gloo/api/v1/enterprise/options/extauth/v1/extauth.proto.sk/#settings" %}}). The external auth settings can be configured on the [global Gloo `Settings` object]({{% versioned_link_path fromRoot="/reference/api/github.com/solo-io/gloo/projects/gloo/api/v1/settings.proto.sk/#settings" %}}).
```
Comment (kdorosh) [inline docs/content/guides/security/auth/extauth/oauth/_index.md:18]: we have the `protobuf` directive for exactly this use case. using it, if we ever move the reference location we wouldn't need to update this link.

sample usage: docs/content/guides/security/rate_limiting/simple/_index.md (see `{{% protobuf name="ratelimit.options.gloo.solo.io.IngressRateLimit" display="format"%}}`)

Comment (lgadban) [reply docs/content/guides/security/auth/extauth/oauth/_index.md:18]: Now I'm wondering if it makes sense to make this a warning a static file so I don't have the content duplicated everywhere...

Comment (kdorosh) [reply docs/content/guides/security/auth/extauth/oauth/_index.md:18]: I also thought about commenting that, probably a good idea too

Comment (lgadban) [reply docs/content/guides/security/auth/extauth/oauth/_index.md]: Unfortunately using the include doesn't seem to work, the static file includes don't parse the `protobuf` directive.

**Pull request merged**